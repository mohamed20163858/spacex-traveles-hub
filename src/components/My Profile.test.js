import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import '@testing-library/jest-dom';
import MyProfile from './My profile';

function TestMyProfile() {
  return (
    <Provider store={store}>
      <MyProfile />
    </Provider>
  );
}
it('renders correctly', () => {
  const tree = renderer
    .create(<TestMyProfile />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('simulate checking a joined mission ', () => {
  // ARRANGE
  render(<TestMyProfile />);

  // ACT

  // Arrange
  const result = screen.getByText('My Missions');
  // ASSERT
  expect(result).toBeInTheDocument();
});
