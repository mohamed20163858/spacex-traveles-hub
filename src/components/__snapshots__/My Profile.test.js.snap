// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="profile-body"
  data-testid="momo-id"
>
  <div
    className="my-missions"
  >
    <h2>
      My Missions
    </h2>
    <table
      className="profile-table"
    >
      <thead />
      <tbody />
      <tfoot />
    </table>
  </div>
  <div
    className="my-rockets"
  >
    <h2>
      My Rockets
    </h2>
    <table
      className="profile-table"
    >
      <thead />
      <tbody />
      <tfoot />
    </table>
  </div>
</div>
`;
